src/isl/util.rs

```diff
@@ -79,33 +91,18 @@ impl Range {
         // set end of the range
         let end = RangeBoundaryValue::from_ion_element(try_to!(range.get(1)), is_non_negative)?;
 
-        // validate both range boundary values
-        let range_type = Range::validate_range_boundaries(&start, &end)?;
-        match range_type {
-            RangeBoundaryValueType::Decimal(_) => Ok(Range::Decimal(start, end)),
-            RangeBoundaryValueType::Float(_) => Ok(Range::Float(start, end)),
-            RangeBoundaryValueType::Integer(_) => Ok(Range::Integer(start, end)),
-            RangeBoundaryValueType::IntegerNonNegative(_) => {
-                Ok(Range::IntegerNonNegative(start, end))
-            }
-            RangeBoundaryValueType::Timestamp(_) => Ok(Range::Timestamp(start, end)),
-        }
+        // validate both range boundary values and returns created `Range`
+        Range::range(start, end)
     }
 
     // helper method to which validates a non negative integer range boundary value
     fn validate_non_negative_integer_range_boundary_value(
         value: &AnyInt,
-    ) -> IonSchemaResult<usize> {
+    ) -> IonSchemaResult<&AnyInt> {
```

Comment:
Does this still need to return a value?

Reply:
No. I can remove it and just return `IonSchemaResult<()>` instead.